Solutions/Branwen/GUI.cs

```diff
@@ -0,0 +1,166 @@
+                folderDialog.Description = "Set Folder to Inventory";
+                if (folderDialog.ShowDialog() != DialogResult.OK)
+                {
+                    return;
+                }
+
+                DirectoryInfo[] topLevelDirectories = new DirectoryInfo(folderDialog.SelectedPath).GetDirectories();
+                string outputFile = Path.Combine(folderDialog.SelectedPath, "MediadriveInventory.xlsx");
+                if (topLevelDirectories == null || outputFile == null)
+                {
+                    MessageBox.Show("This Directory is Invalid. Please try again");
+                    return;
+                }
+
+                if(File.Exists(outputFile))
+                {
+                    File.Delete(outputFile);
+                }
+
+                //Create worksheet
+                SpreadsheetDocument spreadsheetDocument = SpreadsheetDocument.Create(outputFile, DocumentFormat.OpenXml.SpreadsheetDocumentType.Workbook);
+                WorkbookPart workbookpart = spreadsheetDocument.AddWorkbookPart();
+                workbookpart.Workbook = new Workbook();
+                Sheets sheets = spreadsheetDocument.WorkbookPart.Workbook.AppendChild<Sheets>(new Sheets());
+
+                //Loop through all Directories and make a new sheet and put that shit in there
+                for (int i = 0; i < topLevelDirectories.Length; i++)
+                {
+                    WorksheetPart worksheetPart = workbookpart.AddNewPart<WorksheetPart>();
+                    worksheetPart.Worksheet = new Worksheet(new SheetData());
+                    SheetData sheetData = worksheetPart.Worksheet.GetFirstChild<SheetData>();
+                    Sheet sheet = new Sheet();
+                    sheet.Id = spreadsheetDocument.WorkbookPart.GetIdOfPart(worksheetPart);
+                    sheet.Name = topLevelDirectories[i].Name;
+                    sheet.SheetId = (UInt32)Convert.ToInt32(i + 1);
+                    sheets.Append(sheet);
+                    WriteDirectoryToWorksheet(RunInventory(topLevelDirectories[i]), sheetData);
+                }
+                spreadsheetDocument.Close();
+                MessageBox.Show("DONE! Files Inventoried:  " + fileCount);
+                buttonSelectAndRunInventory.Enabled = true;
+                buttonSelectAndRunInventory.Text = "Select Inventory Directory";
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show("Failed to Inventory:" + Environment.NewLine + ex.Message);
+                buttonSelectAndRunInventory.Enabled = true;
+                buttonSelectAndRunInventory.Text = "Select Inventory Directory";
+            }
+        }
+
+        /// <summary>
+        /// Recursively finds all Files in the parent directory
+        /// </summary>
+        /// <param name="parent"></param>
+        /// <returns></returns>
+        private List<FileInfo> RunInventory(DirectoryInfo parent)

```

Comment:
making it IList. IEnumerable and ICollection doesn't give me indexing and Count and Length that I need